client_test.go

```diff
@@ -509,6 +509,38 @@ var helloClient = []string{
 	"NOOP\n",
 }
 
+var shuttingDownServerHello = `220 hello world
+421 Service not available, closing transmission channel
+`
+
+func TestHello_421Response(t *testing.T) {
+	server := strings.Join(strings.Split(shuttingDownServerHello, "\n"), "\r\n")
+	client := "EHLO customhost\r\n"
+	var cmdbuf bytes.Buffer
+	bcmdbuf := bufio.NewWriter(&cmdbuf)
+	var fake faker
+	fake.ReadWriter = bufio.NewReadWriter(bufio.NewReader(strings.NewReader(server)), bcmdbuf)
+	c := NewClient(fake)
+	defer c.Close()
+	c.serverName = "fake.host"
+	c.localName = "customhost"
+
+	err := c.Hello("customhost")
+	if err == nil {
+		t.Errorf("Expected Hello to fail")
+	}
+
+	if err.Error() != "SMTP error 421: Service not available, closing transmission channel" {
```

Comment:
Can we check for an `SMTPError` with `errors.As` here? To make the check more robust.

Reply:
Added the check on https://github.com/emersion/go-smtp/pull/260/commits/0ebd8acee04a66ca8325fbb94895e329d49c4703

Reply:
Added the check on https://github.com/emersion/go-smtp/pull/260/commits/e1cdaa56f2a4b3da47b689426a66ab053df6b7f1